src/omotes_sdk/workflow_type.py

```diff
@@ -842,6 +844,92 @@ def to_pb_value(value: ParamsDictValues) -> float:
     for parameter in WorkflowParameter.__subclasses__()
 }
 
+List["ParamsDictValues"], "ParamsDict", None, float, int, str, bool, datetime, timedelta
+
+
+def check_parameter_relation(
+    value1: ParamsDictValues, value2: ParamsDictValues, check: ParameterRelation
+) -> Literal[True]:
+    """Check if the values adhere to the parameter relation.
+
+    :param value1: The left-hand value to be checked.
+    :param value2: The right-hand value to the checked.
+    :param check: The parameter relation to check between `value1` and `value2`
+    :return: Always true if the function returns noting the parameter relation is adhered to.
+    :raises RuntimeError: In case the parameter relation is not adhered to.
+    """
+    supported_types = (float, int, datetime, timedelta)
+    if not isinstance(value1, supported_types) or not isinstance(value2, supported_types):
+        raise RuntimeError(
+            f"Values {value1}, {value2} are of a type that are not supported "
+            f"by parameter relation {check}"
+        )
+
+    same_type_required = (datetime, timedelta)
+    if (isinstance(value1, same_type_required) or isinstance(value2, same_type_required)) and type(
+        value1
+    ) is not type(value2):
+        raise RuntimeError(
+            f"Values {value1}, {value2} are required to be of the same type to be"
+            f"supported by parameter relation {check}"
+        )
+
+    if check.relation == ParameterRelation.RelationType.GREATER:
+        result = value1 > value2  # type: ignore[operator]
+    elif check.relation == ParameterRelation.RelationType.GREATER_OR_EQ:
+        result = value1 >= value2  # type: ignore[operator]
+    elif check.relation == ParameterRelation.RelationType.SMALLER:
+        result = value1 < value2  # type: ignore[operator]
+    elif check.relation == ParameterRelation.RelationType.SMALLER_OR_EQ:
+        result = value1 <= value2  # type: ignore[operator]
+    else:
+        raise RuntimeError("Unknown parameter relation. Please implement.")
+
+    if not result:
+        raise RuntimeError(
+            f"Check failed for relation {check.relation} with "
+            f"{check.key_1}: {value1} and  {check.key_2}: {value2}"
+        )
+    return result
+
+
+def convert_str_to_parameter_relation(
+    parameter_relation_name: str,
+) -> ParameterRelation.RelationType.ValueType:
+    """Translate the name of a parameter relation to the relevant enum.
+
+    :param parameter_relation_name: String name of the parameter relation.
+    :return: The parameter relation as an enum value of `ParameterRelation.RelationType`
+    :raises RuntimeError: In case the parameter relation name is unknown.
+    """
+    normalized_relation_name = parameter_relation_name.lower()
+    if normalized_relation_name == "greater":
```

Comment:
You could parse directly: `result = ParameterRelation.RelationType.Value(parameter_relation_name.upper())`